earn/src/pages/BoostPage.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  InRangeBadge,
+  OutOfRangeBadge,
+  UniswapPositionCardContainer,
+  UniswapPositionCardWrapper,
+} from '../components/common/UniswapPositionCard';
+import { sqrtRatioToPrice } from '../data/BalanceSheet';
+import {
+  UniswapNFTPosition,
+  UniswapPosition,
+  computePoolAddress,
+  fetchUniswapNFTPositions,
+  getAmountsForLiquidity,
+  tickToPrice,
+} from '../data/Uniswap';
+
+const ACCENT_COLOR = 'rgba(130, 160, 182, 1)';
+
+export default function BoostPage() {
+  const { activeChain } = useContext(ChainContext);
+  const provider = useProvider({ chainId: activeChain.id });
+  const { address: userAddress } = useAccount();
+
+  const [uniswapNFTPositions, setUniswapNFTPositions] = useState<Map<number, UniswapNFTPosition>>(new Map());
+
+  useEffect(() => {
+    let mounted = true;
+    async function fetch() {
+      if (userAddress === undefined) return;
+      const fetchedUniswapNFTPositions = await fetchUniswapNFTPositions(userAddress, provider, activeChain);
+      if (mounted) {
+        setUniswapNFTPositions(fetchedUniswapNFTPositions);
+      }
+    }
+    fetch();
+    return () => {
+      mounted = false;
+    };
+  }, [activeChain, provider, userAddress]);
+
+  const nonZeroUniswapNFTPositions = useMemo(() => {
+    return Array.from(uniswapNFTPositions.values()).filter((position) => {
+      return JSBI.greaterThan(position.liquidity, JSBI.BigInt(0));
+    });
+  }, [uniswapNFTPositions]);
+
+  const contracts = useMemo(() => {
+    return nonZeroUniswapNFTPositions.map((position) => {
+      return {
+        abi: UniswapV3PoolABI,
+        address: computePoolAddress(position),
+        functionName: 'slot0',
+        chainId: activeChain.id,
+      } as const;
+    });
+  }, [activeChain.id, nonZeroUniswapNFTPositions]);
+
+  const { data: slot0Data } = useContractReads({
+    contracts: contracts,
+    allowFailure: false,
+  });
+
+  const uniswapNFTCardInfo = useMemo(() => {
+    if (slot0Data === undefined || slot0Data.length !== nonZeroUniswapNFTPositions.length) {
+      return [];
+    }
+    return nonZeroUniswapNFTPositions.map((position, index) => {
+      const { token0, token1, tickLower, tickUpper } = position;
+
+      const minPrice = tickToPrice(tickLower, token0.decimals, token1.decimals, true);
+
+      const maxPrice = tickToPrice(tickUpper, token0.decimals, token1.decimals, true);
+
+      const sqrtPriceX96 = slot0Data[index][0];
+      const currentTick = slot0Data[index][1];
+
+      const uniswapPosition: UniswapPosition = {
+        lower: tickLower,
+        upper: tickUpper,
+        liquidity: position.liquidity,
+      };
+
+      const [amount0, amount1] = getAmountsForLiquidity(uniswapPosition, currentTick, token0.decimals, token1.decimals);
+
+      const token0PerToken1 = sqrtRatioToPrice(toBig(sqrtPriceX96), token0.decimals, token1.decimals);
+      const amount0InTermsOfToken1 = amount0 * token0PerToken1;
+      const totalValue = amount0InTermsOfToken1 + amount1;
```

Comment:
We have `GoodNumber` ways of doing this, no?

Reply:
I believe this code is taken from other places, but we can look into updating this in the future.